### PR TITLE
More helpful error message for Map(fname) when fname doesn't exist

### LINKED ISSUE #3552: File not found error should include the file path when trying to load Map
Currently trying to load a non-existent file using `Map` raises a very generic error message. It would be good if the file that is trying to be loaded could be listed in the error message.

Current message:

```python
Traceback (most recent call last):
  File "main.py", line 84, in <module>
    aia_map = aia_helpers.create_synoptic_map(datetime.now())
  File "/Users/dstansby/github/psp-connection/aia_helpers.py", line 99, in create_synoptic_map
    aia_synop_map = synop_reproject(aia_map, shape)
  File "/Users/dstansby/github/psp-connection/aia_helpers.py", line 82, in synop_reproject
    new_map = Map(synop_map_path)
  File "/Users/dstansby/miniconda3/lib/python3.7/site-packages/sunpy/map/map_factory.py", line 276, in __call__
    data_header_pairs, already_maps = self._parse_args(*args, **kwargs)
  File "/Users/dstansby/miniconda3/lib/python3.7/site-packages/sunpy/map/map_factory.py", line 238, in _parse_args
    raise ValueError("File not found or invalid input")
ValueError: File not found or invalid input
```

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #3552

- Check if `arg` can be co-erced into a `Path` object.
- If so, step through checking if it is a file, dir, or glob, and use appropriate logic in each case
- Add a test that `arg` is included in the error message when it's not available.

### CODE REVIEW

File: sunpy/map/map_factory.py [hunk 1]
```diff
@@ -231,12 +218,25 @@ def _parse_args(self, *args, **kwargs):
...
+                path = pathlib.Path(arg).expanduser()
+                if _is_file(path):
+                    pairs = self._read_file(path, **kwargs)
+                    data_header_pairs += pairs
+                elif _is_dir(path):
+                    for afile in sorted(path.glob('*')):
+                        data_header_pairs += self._read_file(afile, **kwargs)
+                elif glob.glob(os.path.expanduser(arg)):
+                    for afile in sorted(glob.glob(os.path.expanduser(arg))):
+                        print(afile)
```
Comment (Cadair) [inline sunpy/map/map_factory.py]: remove the print?

**Comment by Cadair** [comment]
Please add a changelog and this looks good to go.

File: sunpy/map/map_factory.py (hunk 1 above)
Comment (dstansby) [reply sunpy/map/map_factory.py]: doh good catch

File: sunpy/map/map_factory.py [hunk 2]
```diff
@@ -350,12 +350,33 @@ def _is_url(arg):
 
 
 def _is_path(arg):
```
Comment (wtbarnes) [inline sunpy/map/map_factory.py]: I think the naming of this function is now misleading. I initially thought the `_parse_args` code above was broken because a glob is not a valid path (though it can be coerced into a `pathlib.Path` object).

File: sunpy/map/map_factory.py (hunk 1 above)
Comment (Cadair) [reply sunpy/map/map_factory.py]: this still needs to go :grinning:

**Comment by Cadair** [comment]
Would you be up for adding a couple of extra tests to cover some of the edge cases here and up the coverage?
